### ISSUE TITLE
Consider using the default MRAN repo for the image instead

### ISSUE THREAD

**Issue by cboettig** [o2r-project/containerit#72]
Really cool project here, and very clever use of the sysreqs package! Just taking a quick look so I may be missing something, but it looks like by default you're setting the Dockerfile to install from `cloud.r-project.org`, rather than letting R stick with the default MRAN snapshot set in the `r-ver` image.  On the latest version (e.g. 3.4.0) the MRAN snapshot date will continue to move each month as the image is rebuilt, but only until the next R version (e.g .3.4.1) is released, after which it will be frozen at the last date 3.4.0 was current.  This means the dockerfile has a much greater likelihood of successfully building the same environment later on.  

To really make sure the package always gets the same version of software, it would be ideal to set the CRAN repo to an MRAN snapshot repo for the current date, e.g. 

```
install2.r -r https://mran.microsoft.com/snapshot/2017-05-30 stringr stringi dplyr 
```

As a side issue, I don't think you'll get a successful install for many packages purely based on sysreqs, unfortunately.  (Maybe I'm wrong -- compare the Dockerfile you get with the spatial packages to the rocker/geospatial dockerfile, for instance.)

**Comment by nuest** [comment]
Thanks for the feedback!

You raise and important point. We set the mirror intentionally to get later CRAN packages, assuming that we would be able to more closely reflect the version a user has on his non-containerized machine, where packages are always up-to-date. That might already be quite an assumption!

Setting the default mirror happens [here](https://github.com/o2r-project/containerit/blob/940a911b9eab4116ac3a7f6d166b648e0bdbbc64/R/package-installation-methods.R#L165)

This is too hidden, so we should

- [ ] do not change the mirror by default
- [ ] make the mirror configurable as part of the `containerit` function, don't hide it in an internal `.variable`. This way we can for our use case use the "latest" CRAN, but provide regular users more stable behaviour
- [ ] maybe even warn if users configure a versioned R base image and then change the CRAN mirror?

**On sysreqs**: Right! Good catch. Our intention is to rely on sysreqs where we can (can contribute to the syseqs database if we discover something), but also put some knowledge into this package where we must deviate from what sysreqs offers.

**Comment by cboettig** [comment]
If the FROM image points to the latest version of R (e.g. `3.4.0`), then you'll be getting pretty close to the latest versions of CRAN packages without changing the default mirror, and you'll have the advantage that the Dockerfile will continue to build successfully even when future packages are released.  

If you want to create an a Dockerfile that will continue to install the latest versions of R packages whenever it is re-built, then you don't want to be locking the R version.  Just use `r-ver:latest` instead of `r-ver:3.4.0` and it will already use RStudio's CRAN CDN to give you the latest packages.  So like you say, it probably makes sense to give a warning if users fix `r-ver:3.4.0` but set the mirror to the latest, since that will almost surely create trouble once newer R versions are released.  

Re `sysreqs`, a possible work-around, though obviously not ideal, is to just set the FROM image to something higher up the rocker stack, e.g. use `rocker/geospatial:3.4.0` if a user is installing the geospatial suite.  (Things will get a little better soon with the Debian stretch release, since we'll have a new stable debian image that will have sufficiently recent binaries for dependencies like gdal or jags so those don't need to be compiled)

**Comment by nuest** [comment]
The function in question `get_docker_cran_mirror()` is gone by now, so we'll simply use the mirror configured in the image.

If users want to use a specific mirror, they may do so by adding the required `RUN` commands manually. _I think this can be closed._
